atlasdb-commons/src/main/java/com/palantir/common/time/NanoClock.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.common.time;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.function.Supplier;
+
+public enum NanoClock implements Supplier<NanoTime> {
```

Comment:
does this necessarily need to be an enum? I know the language allows fields to be mutable, but feels a bit weird. Is it not sufficient to just create a NanoClock as a normal class/object and pass that into `LeaderClock`? you'll still have one per namespace, or if we really want it to span everything, just inject it into `AsyncTimeLockServicesCreator`?